equinix/services/fabricv4/api_client.py

```diff
@@ -312,7 +316,7 @@ def response_deserialize(
                     match = re.search(r"charset=([a-zA-Z\-\d]+)[\s;]?", content_type)
                 encoding = match.group(1) if match else "utf-8"
                 response_text = response_data.data.decode(encoding)
-                return_data = self.deserialize(response_text, response_type)
+                return_data = self.deserialize(response_text, response_type, content_type)
```

Comment:
Fix for issues with deserialization not matching the specified content type: https://github.com/OpenAPITools/openapi-generator/pull/18800